cpmpy/solvers/pysat.py

```diff
@@ -381,48 +413,77 @@ def get_core(self):
-        elif cpm_compsum.name == ">":
-            return CardEnc.atleast(lits=lits, bound=bound+1, vpool=self.pysat_vpool).clauses
-        elif cpm_compsum.name == "==":
-            return CardEnc.equals(lits=lits, bound=bound, vpool=self.pysat_vpool).clauses
-        elif cpm_compsum.name == "!=":
-            # special cases with bounding 'hardcoded' for clarity
-            if bound <= 0:
-                return CardEnc.atleast(lits=lits, bound=bound+1, vpool=self.pysat_vpool).clauses
-            elif bound >= len(lits):
-                return CardEnc.atmost(lits=lits, bound=bound-1, vpool=self.pysat_vpool).clauses
-            else:
-                ## add implication literals for (strict) atleast/atmost, one must be true
-                is_atleast = self.solver_var(boolvar())
-                is_atmost = self.solver_var(boolvar())
-                clauses = [[is_atleast, is_atmost]]
-                clauses += [atl + [-is_atleast] for atl in
-                            CardEnc.atleast(lits=lits, bound=bound+1, vpool=self.pysat_vpool).clauses]
-                clauses += [atm + [-is_atmost] for atm in
-                            CardEnc.atmost(lits=lits, bound=bound-1, vpool=self.pysat_vpool).clauses]
-                return clauses
-
-        raise NotImplementedError(f"Non-operator constraint {cpm_compsum} not supported by CPM_pysat")
+        # unpack and transform to PySAT argument
+        lhs, rhs = cpm_expr.args
+        if lhs.name != "sum":
+            raise NotSupportedError(
+                f"PySAT: Expect {cpm_expr} to be a 'sum'"
+            )
+
+        lits = self.solver_vars(lhs.args)
+        pysat_args = { "lits": lits, "bound": rhs, "vpool": self.pysat_vpool }
+
+        if cpm_expr.name == "<=":
+            return CardEnc.atmost(**pysat_args).clauses
+        elif cpm_expr.name == ">=":
+            return CardEnc.atleast(**pysat_args).clauses
+        elif cpm_expr.name == "==":
+            return CardEnc.equals(**pysat_args).clauses
+
+    def _pysat_pseudoboolean(self, cpm_expr):
+        """ Convert CPMpy comparison of `wsum` (over Boolean variables) into PySAT list of clauses """
+        assert_normalized_bool_lin(cpm_expr)
+        if cpm_expr.args[0].name != "wsum":
+            raise NotSupportedError(
+                f"PySAT: Expect {cpm_expr} to be a 'wsum'"
+            )
+
+        if not CPM_pysat.pb_supported():
+            raise ImportError("Please install PyPBLib: pip install pypblib")
+
+        from pysat.pb import PBEnc
+
+        # unpack and transform to PySAT arguments
+        lhs, rhs = cpm_expr.args
+        lits = self.solver_vars(lhs.args[1])
+        pysat_args = {"weights": lhs.args[0], "lits": lits, "bound": rhs, "vpool":self.pysat_vpool }
+
+        if cpm_expr.name == "<=":
+            return PBEnc.atmost(**pysat_args).clauses
+        elif cpm_expr.name == ">=":
+            return PBEnc.atleast(**pysat_args).clauses
+        elif cpm_expr.name == "==":
```

Comment:
same here: also raise ValueError in else-case

Reply:
+1